src/layers/tile/infinite.rs

```diff
@@ -85,28 +97,64 @@ fn tile_to_chunk_pos(x: i32, y: i32) -> (i32, i32) {
     )
 }
 
-/// Part of an infinite tile layer.
+/// Part of an infinite tile layer's data.
+///
+/// Has only the tile data contained within and not a reference to the map it is part of. It is for
+/// this reason that this type should actually be called `ChunkData`, and so this will change in
+/// the next breaking release. In 99.99% of cases you'll actually want to use [`ChunkWrapper`].
+// TODO(0.11.0): Rename to ChunkData
 #[derive(Debug, PartialEq, Clone)]
 pub struct Chunk {
     tiles: Box<[Option<LayerTileData>; Self::TILE_COUNT]>,
 }
 
 impl Chunk {
-    /// Internal infinite layer chunk width. Do not rely on this value as it might change between
-    /// versions.
+    /// Infinite layer chunk width. This constant might change between versions, not counting as a
+    /// breaking change.
     pub const WIDTH: u32 = 16;
-    /// Internal infinite layer chunk height. Do not rely on this value as it might change between
-    /// versions.
+    /// Infinite layer chunk height. This constant might change between versions, not counting as a
+    /// breaking change.
     pub const HEIGHT: u32 = 16;
-    /// Internal infinite layer chunk tile count. Do not rely on this value as it might change
-    /// between versions.
+    /// Infinite layer chunk tile count. This constant might change between versions, not counting
+    /// as a breaking change.
     pub const TILE_COUNT: usize = Self::WIDTH as usize * Self::HEIGHT as usize;
 
     pub(crate) fn new() -> Self {
         Self {
             tiles: Box::new([None; Self::TILE_COUNT]),
         }
     }
+
+    /// Obtains the tile data present at the position given relative to the chunk's top-left-most tile.
+    ///
+    /// If the position given is invalid or the position is empty, this function will return [`None`].
+    ///
+    /// If you want to get a [`LayerTile`](`crate::LayerTile`) instead, use [`ChunkWrapper::get_tile()`].
+    pub fn get_tile_data(&self, x: i32, y: i32) -> Option<&LayerTileData> {
+        if x < Self::WIDTH as i32 && y < Self::HEIGHT as i32 && x >= 0 && y >= 0 {
```

Comment:
Just wondering, but how useful are these conditions at this low level, given that the main use-case will be iterating the entire chunk as in the example in the docs, making them just a potential performance overhead? Would a debug-only out of bounds check make sense? Of course same consideration applies to fixed tile layers.

Reply:
An unsafe version of the function could probably be supplied as well, `get_tile_data_unchecked`. I agree.

Reply:
It is, most containers (e.g. all std ones) have unchecked unsafe functions to obtain data the user is sure exists.

Reply:
Leaving that for a future PR.